docs/getting-started/config/database.md

```diff
@@ -108,59 +153,269 @@ If you opt for the second option of replacing the entire string, take care to no
+databases named `example_prefix_auth` and `example_prefix_catalog` respectively.
 
-Another way to run Postgres is to use Docker Compose, here's what that would look like:
+```yaml
+backend:
+  database:
+    client: pg
+    connection:
+      host: some.example-pg-instance.tld
+      user: postgres
+      password: password
+      port: 5432
+    prefix: 'example_prefix_'
+```
 
-```yaml title="docker-compose.local.yaml"
-version: '4'
+### Connection Configuration Per Plugin
 
-services:
-  postgres:
-    image: postgres:17.0-bookworm
-    environment:
-      POSTGRES_USER: postgres
-      POSTGRES_PASSWORD: <secret>
-      # If you want to set a timezone you can use the following environment variables, this is handy when trying to figure out when scheduled tasks will run!
-      # TZ: Europe/Stockholm
-      # PGTZ: Europe/Stockholm
-    ports:
-      - 5432:5432
+Both `auth` and `catalog` use connection configuration with different
+credentials and database names. This type of configuration can be useful for
+environments with infrastructure as code or data which may provide randomly
+generated credentials and/or database names.
+
+```yaml
+backend:
+  database:
+    client: pg
+    connection: 'postgresql://some.example-pg-instance.tld:5432'
+    plugin:
+      auth:
+        connection: 'postgresql://fort:knox@some.example-pg-instance.tld:5432/unwitting_fox_jumps'
+      catalog:
+        connection: 'postgresql://bank:reserve@some.example-pg-instance.tld:5432/shuffle_ransack_playback'
 ```
 
-Then you would just run `docker compose -f docker-compose.local.yaml up` to start Postgres.
+#### PostgreSQL and SQLite 3
+
+The example below uses PostgreSQL (`pg`) as the database client for all plugins
+except the `auth` plugin which uses `better-sqlite3`. As the `auth` plugin's client
+type is different from the base client type, the connection configuration for
+`auth` is used verbatim without extending the base configuration for PostgreSQL.
+
+```yaml
+backend:
+  database:
+    client: pg
+    connection: 'postgresql://foo:bar@some.example-pg-instance.tld:5432'
+    plugin:
+      auth:
+        client: better-sqlite3
+        connection: ':memory:'
+```
+
+## Manual Rollback
+
+In some cases you may want to downgrade a migration applied on the database due to some problem, under the hood Backstage uses [Knex](https://knexjs.org/) to handle [migrations](https://knexjs.org/guide/migrations.html) and you can interact direct with Knex interface to **rollback a migration** applied during an upgrade. You can use the `migrate:down` command to rollback a specific migration. You can also use the `migrate:rollback` command to rollback the last batch of migrations. This is necessary because Knex will mark migrations as corrupted if you try to downgrade your Backstage instance without the rollback. Be aware to run those commands in the new version of the Backstage instance, so you can avoid the corrupted migrations for lower versions.
```

Comment:
```suggestion
In some cases you may want to rollback a migration applied on the database, under the hood Backstage uses [Knex](https://knexjs.org/) to handle [migrations](https://knexjs.org/guide/migrations.html) and you can interact direct with Knex interface to **rollback a migration** applied during an upgrade. You can use the `migrate:down` command to rollback a specific migration. You can also use the `migrate:rollback` command to rollback the last batch of migrations. This is necessary because Knex will mark migrations as corrupted if you try to downgrade your Backstage instance without the rollback. Be aware to run those commands in the new version of the Backstage instance, so you can avoid the corrupted migrations for lower versions.
```